Source/WebCore/css/typedom/transform/CSSScale.cpp

```diff
@@ -152,12 +152,16 @@ ExceptionOr<Ref<DOMMatrix>> CSSScale::toMatrix()
     if (!is<CSSUnitValue>(m_x) || !is<CSSUnitValue>(m_y) || !is<CSSUnitValue>(m_z))
         return Exception { TypeError };
 
+    TransformationMatrix matrix { };
+
     auto x = downcast<CSSUnitValue>(m_x.get()).value();
     auto y = downcast<CSSUnitValue>(m_y.get()).value();
     auto z = downcast<CSSUnitValue>(m_z.get()).value();
 
-    TransformationMatrix matrix { };
-    matrix.scale3d(x, y, z);
+    if (is2D())
+        matrix.scale3d(x, y, 1);
+    else
+        matrix.scale3d(x, y, z);
```

Comment:
I think you should either:

1. call `scaleNonUniform(x, y)` in the `if (is2D())` branch and assign `z` in the `else` branch, or
2. check for the 2D case when assigning `z` to either `1` or its actual value so you only call `scale3d()`.